### DISCUSSION TITLE
No connections found

### DISCUSSION THREAD

**Discussion by dkJohan** [codefori/vscode-ibmi#1285] [Q&A] [answered]
Hi
I've been using code for ibm i  (CFI) for years without any problems and great joy btw.
Last week though, CFI stopped finding my connections and stopped me from adding new ones.
<img width="219" alt="image" src="https://github.com/halcyon-tech/vscode-ibmi/assets/36447157/42d31f63-fa36-4df3-bf60-7aa7a96b76ac">
The blue line, just keeps running from left to right in my interface and I'm not able push or use hot keys in this pane to create a new connection.
What I've done and tried so far: 
- confirmed that my collegaues CFI works fine
- confirmed that a connection is in the settings
- removed connections from settings - did not help
-  removed all (like in all all) extensions from VS including settings - reinstalled CFI development pack - did not help
- removed settings for CFI, CFI and VS Code - reinstalled VS and CFI - did not help

Now I'm out of ideas. 
Can any one help me out here, so I can be productive again 
regards Johan

**Comment by chrjorgensen** [comment]
@dkJohan

Please provide more debug information - best way is to click the `Report an issue` in the `HELP AND SUPPORT` view and copy-paste the generated text from the temp issue to this issue.

Also provide the VS Code version and the Code for IBM i version.

**Reply by dkJohan** [reply]
Extension version: 1.8.0
VS Code version: Code 1.78.2 (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:39:26.248Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.75GB (5.76GB free)|
|Process Argv|--crash-reporter-id e0d3834e-28d0-48d8-b85b-e9f0b0b85ac5|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaac:30438847
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30610678
282f8724:30602487
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30705553
azdwalk:30721579
pythonms35:30701012
ecj1e332:30736112
pythonfmttext:30731395
fixhidewlkth:30730051
showsbindicator:30730056
pythongtdpathcf:30739705

```

</details>

**Reply by dkJohan** [reply]
FYI A colleague has the same problem now (after upgrading).

**Reply by chrjorgensen** [reply]
@dkJohan What was upgraded by your colleague? VS Code or Code for IBM i?

**Reply by chrjorgensen** [reply]
@dkJohan FWIW, I also had a problem last week after an upgrade: I use [VS Codium](https://vscodium.com/) and there was a problem with the latest build due to a [GitHub failure](https://github.com/VSCodium/vscodium/issues/1515). I managed to get around this by downgrading to the previous version of VS Code and stop it from updating. And today I upgraded to the newest version after the issue was fixed.

If your problem only occurs on VS Code 1.78.2, I would recommend you try a downgrade of VS Code to 1.77.3 - to see if that solves the problem...

**Reply by dkJohan** [reply]
> @dkJohan What was upgraded by your colleague? VS Code or Code for IBM i?

code for ibm i

**Comment by chrjorgensen** [comment]
@dkJohan 

Also provide any output from the `Debug console`...

**Reply by dkJohan** [reply]
No output from debug console. Be aware that we use source code from liberarys   eg. mylib/qsrc.

**Comment by sebjulliand** [comment]
Hi @dkJohan , would you mind sharing your user settings as JSON, specifically these two arrays:
- "code-for-ibmi.connections"
- "code-for-ibmi.connectionSettings"

Thanks!

**Reply by sebjulliand** [reply]
Also, click once on `Connect to an IBM i` and then:
- Press `ctrl+shift+i`
- Open the `Console` tab
- Copy and paste the whole `Console` content here

![image](https://github.com/halcyon-tech/vscode-ibmi/assets/11096890/fa2d47be-57ed-484a-ba50-db8961fc7947)

**Reply by dkJohan** [reply]
It's not possible for me to press any key or option:
when I click the server icon it shortly says "activating extensions..." and though it goes away I'm not able to press the "connect.." button or use any short cut keys.

**Reply by dkJohan** [reply]
> Hi @dkJohan , would you mind sharing your user settings as JSON, specifically these two arrays:
> 
> * "code-for-ibmi.connections"
> * "code-for-ibmi.connectionSettings"
> 
> Thanks!

Before reinstal of VS Code it looked like this:
{
    "code-for-ibmi.connectionSettings": [
        {
            "name": "ktp01",
            "sourceFileList": [
                "WO001WBO/QSRC",
                "WB001COMIE/QSRC"
            ],

and 

  "code-for-ibmi.connections": [

    
        {
            "name": "ktp01",
            "host": "ktp01",
            "port": 22,
            "username": "changed_to_this",
            "privateKey": null,
            "password": "changed_to_this",
            "keepaliveInterval": 35000
        }
    ],

after reinstall:

    "code-for-ibmi.connectionSettings": [
        
    
    ],
    "code-for-ibmi.connections": []
}

**Reply by dkJohan** [reply]
We did some change to our network last week. Is it possible, that this is related to this? I can start ssh from 5250 and from PUTTY though.

**Reply by sebjulliand** [reply]
> It's not possible for me to press any key or option: when I click the server icon it shortly says "activating extensions..." and though it goes away I'm not able to press the "connect.." button or use any short cut keys.

OK so open this Console before clicking on the Connect button and then click on it. Hopefully there will be some output 😅

**Reply by chrjorgensen** [reply]
@dkJohan @sebjulliand Please see my [comment above](https://github.com/orgs/halcyon-tech/discussions/1285#discussioncomment-5904329
) - I suspect VS Code is causing this issue...

**Reply by dkJohan** [reply]
it's not possible to click the button or use the hot key at all.

**Reply by sebjulliand** [reply]
How about opening the developer tools right after VScode started?
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/11096890/a769bb51-0472-4edb-8095-613f4439d1b7)

**Reply by chrjorgensen** [reply]
@dkJohan @sebjulliand I'll post in this thread so we can keep the communication close...

Since this issue started after an upgrade, I would recommend trying a downgrade - either of Code for IBM i and/or VS Code.

Code for IBM i 1.8.0 was released almost a month ago, and we have not seen any other issues similar to this - but we have previously seen VS Code introduce an undocumented change, that broke Code for IBM i. That's why I also suggest a downgrade of VS Code.

I have just upgraded my VS Code instance to 1.78.2 and have Code for IBM i 1.8.0 running without any issues. So this must be a special case...

**Reply by dkJohan** [reply]
that works and it gave me a hint
<img width="317" alt="image" src="https://github.com/halcyon-tech/vscode-ibmi/assets/36447157/87438d4b-e22c-439d-b1f7-7c73c089ef1b">

which lead me to this
https://github.com/microsoft/vscode/issues/182070

I will dig into this and give my response here, when I know more

**Reply by dkJohan** [reply, marked as answer]
okay - i'm up and running :-).
Thank You both for pointing me in the right direction.

The above host "ktp..." was used to acces source members stored in txt files. I dropped this connection long time ago. I suspect that some reference to this is hidden somewhere.

My solution was to add the host to the  security.allowedUNCHosts setting


<img width="453" alt="image" src="https://github.com/halcyon-tech/vscode-ibmi/assets/36447157/e2da7ede-7fff-4126-bef3-e1a4ef1db6d0">

**Reply by dkJohan** [reply]
read more here
https://code.visualstudio.com/docs/setup/windows#_working-with-unc-paths

**Reply by chrjorgensen** [reply]
@dkJohan Glad to hear you made it work! 😃 

I'm just curious why you access streamfile sources using UNC - was it a Windows fileserver or an IBM i NetServer? Code for IBM i can access IFS files, so using NetServer has never been a requirement. My source files are placed in the IFS since 2016...

**Reply by dkJohan** [reply]
A rather large part of our source files are located on the IFS and has been since 2005.  
To be honest I do not know why "ktp..." ever has been connected using UNC from my pc. 
One of my collegues had the same issue, and another did not. 
I think we will have to check settings on the server and in our network to find out more.

**Reply by chrjorgensen** [reply]
> A rather large part of our source files are located on the IFS and has been since 2005.

Nice! I hope you will continue enjoying Code for IBM i.
